Scripts/Runtime/Entities/PersistentData/PersistentDataSystem.cs

```diff
@@ -0,0 +1,68 @@
+using Anvil.CSharp.Collections;
+using System;
+using System.Collections.Generic;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    internal partial class PersistentDataSystem : AbstractAnvilSystemBase
+    {
+        private static readonly Dictionary<Type, AbstractPersistentData> s_ThreadPersistentData = new Dictionary<Type, AbstractPersistentData>();
+        private static int s_InstanceCount;
+
+        private readonly Dictionary<Type, AbstractPersistentData> m_EntityPersistentData;
+        
+        public PersistentDataSystem()
+        {
+            s_InstanceCount++;
+            m_EntityPersistentData = new Dictionary<Type, AbstractPersistentData>();
+        }
+
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+            Enabled = false;
+        }
+
+        protected override void OnDestroy()
+        {
+            m_EntityPersistentData.DisposeAllValuesAndClear();
+            s_InstanceCount--;
+            if (s_InstanceCount <= 0)
+            {
+                s_ThreadPersistentData.DisposeAllValuesAndClear();
+            }
+            base.OnDestroy();
+        }
+
+        protected override void OnUpdate()
+        {
+            //Just in case
+            Enabled = false;
```

Comment:
Why does this system need to participate in updates? Are there queries created against it? Seems like this should just be a data system.

Reply:
Yep, good point. I forgot we had those. Changed to data system.